### PR TITLE
s3 role and version

### PR DESCRIPTION
**Pull request by anijra**
- add Version in params AWS 
- add possibility used roles AWS and not AccessKeyID/SecretAccessKey

### CODE REVIEW

File: classes/dfs/amazon.php [hunk 1]
```diff
@@ -36,13 +36,16 @@ public static function build()
...
         $parameters = $ini->group( "BackendSettings" );
 
-        $options = array(
-            'credentials' => [
-                'key' => $parameters['AccessKeyID'],
-                'secret' => $parameters['SecretAccessKey'],
-            ],
-            'version' => 'latest'
-        );
+        if(isset($parameters['AccessKeyID'], $parameters['SecretAccessKey'])) {
```
Comment (andrerom) [inline classes/dfs/amazon.php]: Hi, thanks for contributing!

Q: Why did you add the` isset` here when you did not for `Version`?

Comment (anijra) [reply classes/dfs/amazon.php]: Hi, 

Credentials and Version are not related. Version is mandatory. But credentials are sent only if they are set. When you are using AWS role, credentials should not be used.

File: settings/dfsamazons3.ini [hunk 2]
```diff
@@ -7,3 +7,4 @@ Bucket=
 # Your bucket's region, as one  of the regions  on  https://github.com/ezsystems/ezdfs-fsbackend-aws-s3/blob/master/README.md#configuration
 # example: eu-west-1
 Region=
+Version=
```
Comment (andrerom) [inline settings/dfsamazons3.ini]: Would this be correct here perhaps?
```suggestion
# [Optional] version: (string, required) The version of the webservice to utilize (e.g., 2006-03-01).
Version=
```

File: classes/dfs/amazon.php (hunk 1 above)
Comment (andrerom) [reply classes/dfs/amazon.php]: ok, so then maybe change this line to the following to make it more clear you are checking if it has value or not:
```suggestion
        // Only set credentials if configured in INI, if not client will attempt to load them from environment.
        if($parameters['AccessKeyID'] && $parameters['SecretAccessKey']) {
```
And maybe add in the ini something like the following before AccessKeyID:
```
# These maps to client credentials "key" & "secret". If not set, the SDK will attempt to load them from the environment.
```

Also, should default value of `Version` in INI be set to `latest` like before? If this is implicit the same as empty, then let's add that to the suggestion for Version INI doc as well.

**Comment by andrerom** [comment]
@anijra Looking a bit more on this when it comes to "version".

Why do you want to make this configurable? I would believe it should be hard coded to `2006-03-01` as the code here will need to be adjusted if it changes, right?

So given you make credentials optional, is it enough to move the version code outside your new `if` instead?
For example just simply: `$options['version'] = '2006-03-01';`

ref:
https://docs.aws.amazon.com/aws-sdk-php/v3/api/class-Aws.AwsClient.html#___construct
https://docs.aws.amazon.com/AmazonS3/latest/API/Welcome.html

**Comment by anijra** [comment]
@andrerom my goal is to use the "version" that is configured in ezplatform with the bundle "league/flysystem-aws-s3-v3" and to be able to inject it into legacy with an event "LegacyEvents :: PRE_BUILD_LEGACY_KERNEL"
https://doc.ezplatform.com/en/latest/guide/clustering_aws_s3/#toc

**Comment by andrerom** [comment]
@anijra ok clear, then I took the liberty to do the following changes:
- Set default version to `latest` like before, and like suggested in platform doc
- Add additional inline doc in ini and readme


Looks ok to you? If you are fine with this now and it works for you, then I think we can merge it.

**Comment by andrerom** [comment]
@anijra ok clear, then I took the liberty to do the following changes:
- Set version to `latest` like before, and like suggested in platform doc
- Add additional inline doc in ini and readme


Looks ok to you? If you are fine with this now and it works for you, then I think we can merge it.

**Comment by anijra** [comment]
@andrerom it's good for me, Thank you

**Comment by andrerom** [comment]
Thanks for the contribution @anijra!
